api/src/main/java/com/datatorrent/api/Module.java

```diff
@@ -21,8 +21,136 @@
 import org.apache.hadoop.classification.InterfaceStability;
 import org.apache.hadoop.conf.Configuration;
 
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.Operator.InputPort;
+import com.datatorrent.api.Operator.OutputPort;
+import com.datatorrent.api.Operator.Unifier;
+
+/**
+ * A Module is a component which can be added to the DAG similar to the operator,
+ * using addModule API. The module should implement populateDAG method, which
+ * will be called by the platform, and DAG populated by the module will be
+ * replaced in place of the module.
+ *
+ */
 @InterfaceStability.Evolving
 public interface Module
 {
   void populateDAG(DAG dag, Configuration conf);
+
```

Comment:
This allows module to inspect other properties, This will allow modules to implement additional functionality depending on global properties. for example if schema definition is common across all modules in the application, then a single property for schema can be read by all modules and configure operator accordingly.